### PR TITLE
gettext: remove dead patch causing failures on BSDs

### PR DESCRIPTION
**Pull request by rhelmot**
The upgrade to 0.22.5 did not remove this patch slated for removal in 0.22.5, breaking the FreeBSD build.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd (cross)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/gettext/default.nix [hunk 1]
```diff
@@ -19,16 +19,7 @@ stdenv.mkDerivation rec {
...
-  ]
-    # An accidental inclusion in https://marc.info/?l=glibc-alpha&m=150511271003225&w=2
-    # resulted in a getcwd prototype to be added in when it isn't needed.
-    # Clang does not like this unless the "overridable" attribute was appied.
-    # Since we don't need to redeclare getcwd, we can just remove it.
-    #
-    # Issue: https://github.com/NixOS/nixpkgs/issues/348658
-    # Fixed in https://github.com/autotools-mirror/gettext/commit/cb2c1486336462c8180f487221181ee798b0e73e
-    # Remove in 0.22.5 upgrade.
-    ++ lib.optional (stdenv.cc.isClang && !stdenv.targetPlatform.isDarwin) ./fix-getcwd-clang.patch;
```
Comment (paparodeo) [inline pkgs/development/libraries/gettext/default.nix:31]: the patch file can be removed as well.

Comment (rhelmot) [reply pkgs/development/libraries/gettext/default.nix:31]: You're so right!

**Review by paparodeo — APPROVED** [review]
thanks for fixing.

**Comment by Artturin** [comment]
This should have gone to staging instead of master, now it doesn't build on master because the bump went to staging https://nixpk.gs/pr-tracker.html?pr=358254

**Comment by rhelmot** [comment]
Anything I can do to help fix the mess?

**Comment by DavidTruby** [comment]
I wonder if this can be reverted and re-merged onto staging instead, if there's nothing we can do to get gettext onto master faster? pkgsLLVM is still broken on Linux because of this

**Comment by paparodeo** [comment]
> I wonder if this can be reverted and re-merged onto staging instead, if there's nothing we can do to get gettext onto master faster? pkgsLLVM is still broken on Linux because of this

it is already in master

**Comment by DavidTruby** [comment]
I must have missed it by a few seconds; I had checked again just before I wrote that comment!
I can confirm everything works now. Thanks :)

**Pull request merged**